tests/reporters.test.ts

```diff
@@ -47,7 +47,14 @@ describe("JestReporter", () => {
             expect(fs.existsSync(STATE_FILE)).toBeTruthy();
 
             const serialized = JSON.parse(fs.readFileSync(STATE_FILE, "utf-8"));
-            expect(serialized).toEqual({});
+            if(process.env.KELONIO_KEEP_STATE_AT_START === "true") {
```

Comment:
Can you expand on this unit test to test the two behaviours? I envisage a unit test which tests it with behaviour A and behaviour B. This test would probably best just override the environment variable to always force one behaviour as it used to be.

Reply:
That is true, they should be separate tests. I will split these and override the environment variables

Reply:
Addressed in a new commit